server/socket/handlers/handleMove.js

```diff
@@ -116,13 +135,31 @@ const handleMove = (io, socket, rooms, gameSchema, gameModel) => (roomNumber, fr
         const currentFen = game.getCurrentFEN();
         const validMoves = game.validMoves();
         const history = game.getGameHistory();
-        io.to(roomNumber).emit(EVENTS.MOVE_MADE, to, currentFen, validMoves, history);
+        // io.to(roomNumber).emit(EVENTS.MOVE_MADE, to, currentFen, validMoves, history);
+
+        const room = rooms[roomNumber];
+        room.players.forEach(player => {
+            io.to(player.id).emit(EVENTS.MOVE_MADE, to, currentFen, validMoves, history);
+        });
+        room.spectators.forEach(spectator => {
```

Comment:
same for this

Reply:
Is there anything wrong here?

Reply:
just wondering why we have events to spectators. if they're of use then let them be. if they dont serve a purpose remove them as it would be bad code otherwise

Reply:
Because the original code io.to(roomNumber) does not work but Dan mentioned that client sends the room number as number while the server only accepts the room number as string. However in order to do the something similar, I io.to(players) and io.to(spectators).

Reply:
I tried the code Dan fixed but it still does not work, so I still used this code.

Reply:
okay if it works then i'm okay with it

Reply:
i think this could've been coded differently but in the end it works. Spectator just gets treated as player 3 but can't move pieces.

Reply:
at this point, it shouldnt be a problem. I can't think of an event that happens that would cause spectator to bug out. Once the game ends, the room practically closes for both players and all specs.